DeviceSDK/src/main/java/org/astarteplatform/devicesdk/protocol/AstarteServerPropertyInterface.java

```diff
@@ -52,9 +55,10 @@ public AstarteServerValue build(String interfacePath, Object serverValue, DateTi
               .build();
     } else {
       // Couldn't find the mapping
-      System.err.printf(
-          "Got an unexpected path %s for interface %s!%n", interfacePath, getInterfaceName());
-
+      logger.log(
+          Level.SEVERE,
+          String.format(
```

Comment:
Is there any reason why here you've decided to keep the `String.format` call ?
The string concatenation is more readable and widely used in Java, i think this could just be written as:
```java
logger.severe("Got an unexpected path '" + intefacePath + "' for interface " + getInterfaceName());
```
I've changed a bit the message logged it shouldn't be a problem imo but we can keep it as is if necessary.

Reply:
Thanks for the suggestions. I made changes.